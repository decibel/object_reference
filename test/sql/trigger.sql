\set ECHO none

\i test/load.sql

CREATE TABLE test_table();
CREATE TRIGGER test_trigger
  AFTER INSERT OR UPDATE
  ON test_table
  FOR EACH ROW EXECUTE PROCEDURE suppress_redundant_updates_trigger()
;


SELECT plan(
  0
  +3 -- initial
);

SELECT lives_ok(
  $$CREATE TEMP TABLE test_object AS SELECT object_reference.object__getsert('trigger', 'test_trigger') AS object_id;$$
  , $$CREATE TEMP TABLE test_object_id AS SELECT object_reference.object__getsert('trigger', 'test_trigger') AS object_id;$$
);
SELECT is(
  (SELECT object_oid FROM _object_reference.object WHERE object_id = (SELECT object_id FROM test_object))
  , (SELECT oid FROM pg_trigger WHERE tgname = 'test_trigger' AND tgrelid = 'test_table'::regclass)
  , 'Verify regclass field is correct'
);
SELECT is(
  object_reference.object__getsert('trigger', 'test_trigger')
  , (SELECT object_id FROM test_object)
  , 'Existing object works, provides correct ID'
);

-- TODO: Test renaming of trigger

\i test/pgxntool/finish.sql

-- vi: expandtab sw=2 ts=2
