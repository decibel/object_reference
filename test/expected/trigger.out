\set ECHO none
1..3
not ok 1 - CREATE TEMP TABLE test_object_id AS SELECT object_reference.object__getsert('trigger', 'test_trigger') AS object_id;
# Failed test 1: "CREATE TEMP TABLE test_object_id AS SELECT object_reference.object__getsert('trigger', 'test_trigger') AS object_id;"
#     died: 42P01: relation "test_trigger" does not exist
#         CONTEXT:
#             PL/pgSQL function object_reference.object__getsert_w_group_id(cat_tools.object_type,text,text,integer) line 56 at RAISE
#             SQL statement "CREATE TEMP TABLE test_object AS SELECT object_reference.object__getsert('trigger', 'test_trigger') AS object_id;"
#             PL/pgSQL function lives_ok(text,text) line 14 at EXECUTE
ERROR:  42P01: relation "test_object" does not exist
LINE 2: ...e.object WHERE object_id = (SELECT object_id FROM test_objec...
                                                             ^
LOCATION:  parserOpenTable, parse_relation.c:1160
